### DISCUSSION TITLE
creating Windows 24 bits icons (for freewrap) incorrectly produce 32 bits

### DISCUSSION THREAD

**Discussion by mmtsuchi-fr** [ImageMagick/ImageMagick#2952] [Development]
Hi,
I tried to make icons that comply with [http://freewrap.sourceforge.net/](url), so as to say : 
 - BMP Format
 - Size     | Colors
- 16x16  | 16
- 16x16  | 256
- 32x32    | 4-bit
- 32x32    | 8-bit
- 32x32    | 24-bit
- 48x48      | 4-bit
- 48x48      | 8-bit
- 48x48      | 24-bit
- 128x128    | 24-bit
- 256x256      | 24-bit

I tried

> convert input.jpg -verbose \
> \( -clone 0 -compress none -resize 16x16! 	-type PaletteAlpha   -colors 16 -depth 4  -format bmp3 \) \
> \( -clone 0 -compress none -resize 16x16! 	-type PaletteAlpha   -colors 256 -depth 8 -format bmp3 \) \
> \( -clone 0 -compress none -resize 32x32! 	-type PaletteAlpha   -colors 16 -depth 4  -format bmp3 \) \
> \( -clone 0 -compress none -resize 32x32! 	-type PaletteAlpha   -colors 256 -depth 8  -format bmp3 \) \
> \( -clone 0 -compress none -resize 32x32! 	-type TrueColor      -depth 24    -define bmp:format=bmp3 -alpha off \) \
> \( -clone 0 -compress none -resize 48x48! 	-type PaletteAlpha   -colors 16 -depth 4  -format bmp3 \) \
> \( -clone 0 -compress none -resize 48x48! 	-type PaletteAlpha   -colors 256 -depth 8  -format bmp3 \) \
> \( -clone 0 -compress none -resize 48x48! 	-type TrueColor      -depth 24   -define bmp:format=bmp3 -alpha off \) \
> \( -clone 0 -compress none -resize 128x128! -type TrueColor      -depth 24   -define bmp:format=bmp3 -alpha off \) \
> \( **-compress none -clone 0 -resize 256x256! -type TrueColor      -depth 24   -define bmp:format=bmp3 -alpha off** \) \
> -delete 0 \
> out.ico

_(edited : reformated for this post)_

By analysing in hexadecimal, resulting .ico contains **unexpectively 32-bits depth instead of 24-bits.** icon
**Incorrectly produces TrueColorAlpha instead of expected TrueColor icons.**
(other options are consequents of try & see)

versions : 
- ImageMagick 6.9.10-23 Q16 x86_64 20190101
- ImageMagick 7.0.10-44 Q16 x86_64

Best regards

**Comment by jsummers** [comment]
Looking at the source code, I think that ImageMagick has no such feature (as of v7.0.10-48).

Oddly, it does seem to have _some_ code for ICO bit depth 24, but the code might be unreachable, and I'm not sure it would work quite right anyway.

Just to be clear about this, "24-bit" ICO icons always have a 1-bit transparency mask -- they're really 25 bits/pixel, R8G8B8A1.
